mopidy/core/history.py

```diff
@@ -28,11 +28,21 @@ def _add_track(self, track):
         name_parts = []
         if track.artists:
             name_parts.append(
-                ", ".join(artist.name for artist in track.artists)
+                ", ".join(artist.name for artist in track.artists if artist.name)
```

Comment:
Or, rather than all that stuff below you could do something like:
```
artists_names = [artist.name for artist in track.artists if artist.name]
if artists_names:
    name_parts.append(", ".join(artists_names))
```
Which leaves `name_parts` empty if there are no artists with names, and then the original `name = " - ".join(name_parts)` works as required.